sensors/cmd/start.go

```diff
@@ -11,35 +12,17 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
 
-	argoevents "github.com/argoproj/argo-events"
 	"github.com/argoproj/argo-events/common"
 	"github.com/argoproj/argo-events/common/logging"
 	"github.com/argoproj/argo-events/metrics"
 	eventbusv1alpha1 "github.com/argoproj/argo-events/pkg/apis/eventbus/v1alpha1"
 	v1alpha1 "github.com/argoproj/argo-events/pkg/apis/sensor/v1alpha1"
 	"github.com/argoproj/argo-events/sensors"
+	"github.com/fsnotify/fsnotify"
 )
 
 func Start() {
 	logger := logging.NewArgoEventsLogger().Named("sensor")
-	kubeConfig, _ := os.LookupEnv(common.EnvVarKubeConfig)
-	restConfig, err := common.GetClientConfig(kubeConfig)
-	if err != nil {
-		logger.Fatalw("failed to get kubeconfig", zap.Error(err))
-	}
-	kubeClient := kubernetes.NewForConfigOrDie(restConfig)
-	encodedSensorSpec, defined := os.LookupEnv(common.EnvVarSensorObject)
-	if !defined {
-		logger.Fatalf("required environment variable '%s' not defined", common.EnvVarSensorObject)
-	}
-	sensorSpec, err := base64.StdEncoding.DecodeString(encodedSensorSpec)
-	if err != nil {
-		logger.Fatalw("failed to decode sensor string", zap.Error(err))
-	}
-	sensor := &v1alpha1.Sensor{}
-	if err = json.Unmarshal(sensorSpec, sensor); err != nil {
-		logger.Fatalw("failed to unmarshal sensor object", zap.Error(err))
-	}
 
 	busConfig := &eventbusv1alpha1.BusConfig{}
 	encodedBusConfigSpec := os.Getenv(common.EnvVarEventBusConfig)
```

Comment:
Shouldn't the eventbus config be something we get every time on restart. It's possible it could change. Same with the eventbus subject and hostname.

Reply:
If so, we would need to pass this information in the configmap as well

Reply:
We could pass this information via configmap as well. Not sure if that would be necessary, I'm guessing this information changes less frequently so the current mechanism (with a restart) should likely suffice?

Reply:
true.. someone else can commit that in the future if needed.